App.js

```diff
@@ -306,25 +313,41 @@ export default class App extends React.Component {
             linking={linking}
             theme={navTheme}
             onStateChange={(state) => {
-              console.log('STATE CHANGED: ', state);
+              const lastChatIndex = _.findLastIndex(state.routes, {name: 'Chat'});
+              const lastRHPIndex = _.findLastIndex(state.routes, (route) => route.name !== 'Chat' && route.name !== 'LeftHandNav');
```

Comment:
In App we could make this cleaner with some helper methods like `isChatRoute` or `isLHNRoute`. I guess we cant get  rid of these comparisons

Reply:
+1 except I think there must be some way to store something on the route or screen so that a route can define itself as "sidebar", "main", "right hand panel" etc.